### ISSUE TITLE
Option to remove/redact information from logs (hostname, directories)

### ISSUE THREAD

**Issue by repligator** [MegaMek/megamek#5844]
In my opinion, it would be nice to be able to remove/redact a couple of potential sources of personal information in the logs. Specifically the host-name of the computer being used and the directory MM/MHQ/MML is being run from/files loaded from. For example:

`megamek.server.Server.<init>(Server.java:382) - s: hostname = 'John Does Macbook' port = 2346`
would be changed to
`megamek.server.Server.<init>(Server.java:382) - s: hostname = 'REDACTED-HOSTNAME' port = 2346`

`mekhq.gui.dialog.DataLoadingDialog$Task.doInBackground(DataLoadingDialog.java:345) - Loading campaign file from XML file John/mekhq/MekHQ/campaigns/Autosave-4-My Campaign-30360718.cpnx.gz`
would be changed to
`mekhq.gui.dialog.DataLoadingDialog$Task.doInBackground(DataLoadingDialog.java:345) - Loading campaign file from XML file REDACTED.../Autosave-4-My Campaign-30360718.cpnx.gz`

**Comment by HoneySkull** [comment]
I second this.  Either a flag to tell mm not to put this in the logs, or some kind of log cleansing utility to clean it all out before posting.

**Comment by IllianiCBT** [comment]
Yeah, I'll add another vote for this. I did take a look myself, but I don't know my way around the code well enough to resolve this confidently.

**Comment by Thom293** [comment]
It appears username and some paths were also saved in at least one of the new logs created on my system during a crash the other night.  Although I am not an expert at reading such things.
